### ISSUE TITLE
Error: mysql2 connection hangs indefinitely after error in execute()

### ISSUE THREAD

**Issue by marcakafoddex** [sidorares/node-mysql2#1185]
## Description
When executing a prepared statement fails, the next query on that same connection hangs indefinitely. I am encountering this problem in my project when I try to do the following:

- get a connection from the pool
- start a transaction
- execute prepared statements
- commit on success, rollback on failure (ie. exception)
- release the connection

Now when I try to rollback on failure because I make a mistake, let's say I have a syntax error in my SQL, or I put an invalid value (object reference) in a field, an exception is thrown, but then when I try to execute the ROLLBACK query afterwards it hangs indefinitely.

## Code that causes the bug
```JavaScript
const mysql = require('mysql2');
const pool = mysql.createPool({
  host: 'localhost',
  user: 'root',
  password: 'XXX',
  database: 'XXX',
});
class MyClass {}
const mysql = require('mysql2');
const pool = mysql.createPool({
  host: 'localhost',
  user: 'root',
  password: '_chantie12',
  database: 'accountassist',
});
class MyClass {}
pool.getConnection((err, conn) => {
	conn.promise().query('START TRANSACTION')
		.then(() => {
			console.log('Started transaction, going to execute my queries...');
			conn.promise().execute('SELECT ?', [new MyClass()])
				.catch(() => {
					console.log('Oh no, caught error, executing rollback...');
					conn.promise().query('ROLLBACK')
						.catch(() => {
							console.log('... rollback failed!');
						})
						.then(() => {
							console.log('... rollback complete!');
							pool.releaseConnection(conn);
						});
				});
		})
});

```

What I **expect** to see when I execute this, is:
```
Started transaction, going to execute my queries...
Oh no, caught error, executing rollback...
... rollback complete!
```
But what I **see** is this:
```
Started transaction, going to execute my queries...
Oh no, caught error, executing rollback...
```
In other words, the ROLLBACK query call hangs indefinitely. It never times out, or comes back in any way.

## Hint to possible source of the problem
The bug seems related to handling prepared statements. When I try the above with a regular query, instead of a prepared statement (and ofc. not using the ? format, but causing a problem in another way, eg. having invalid syntax), everything works exactly as expected, and the follow up query executes just fine!
